docs/book/src/reference/submodule-layouts.md

```diff
@@ -0,0 +1,243 @@
+# Sub-Module Layouts
+
+This part describes how to modify a scaffolded project for use with multiple `go.mod` files for APIs and Controllers.
+
+Sub-Module Layouts (in a way you could call them a special form of [Monorepo's][monorepo]) are a special use case and can help in scenarios that involve reuse of APIs without introducing indirect dependencies that should not be available in the project consuming the API externally.
+
+See [Using an external Type](/reference/using_an_external_type.md) for more info on how to consume Types and APIs from other projects externally.
+
+## Overview
+
```

Comment:
Could we also add here a WARNING with something like:

Note that you can create controllers to reconcile types that are onwed and defined for other projects or by Kubernetes API itself. If that is the reason for you are looking at this doc please ensure that you check the legacy doc [Using external types](https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/using_an_external_type.md ). 

Currently, the tool is not handle this scenarios automatically and you will need to perform some changes manually.  For further information see the [issue 1999](https://github.com/kubernetes-sigs/kubebuilder/issues/1999) opened to address this need. On top of that, be aware that you will need to add the schema for the external type and would be recommended ensure that the API is applied on the cluster when you are initializing the manager (main.go). You might find beneficial give a look in this [blog](https://developers.redhat.com/blog/2020/02/04/how-to-use-third-party-apis-in-operator-sdk-projects#step_4__use_the_api_in_the_controllers)

Reply:
I think the warning would then replace my comment from line 7. I will try to find better wording and put it in the docs. Regarding the API ensurance on the cluster there is not really an option for that for internally created types as well and one has to run make install in this case. I really think that this is not something the guide here should make note of since if you are using an external API I kind of expect this.